client/i18n.js

```diff
@@ -1,26 +1,39 @@
 import i18n from 'i18next'
-import LanguageDetector from 'i18next-browser-languagedetector'
+import LngDetector from 'i18next-browser-languagedetector'
 import { reactI18nextModule } from 'react-i18next'
 import en from 'locales/en/translation.json'
 import ja from 'locales/ja/translation.json'
 
-i18n
-  .use(LanguageDetector)
-  .use(reactI18nextModule)
-  .init({
-    fallbackLng: 'en',
-    debug: true,
-    interpolation: { escapeValue: false },
-    react: {
-      wait: false,
-      bindI18n: 'languageChanged loaded',
-      bindStore: 'added removed',
-      nsMode: 'default',
-    },
-    resources: {
-      en: { translation: en },
-      ja: { translation: ja },
+export default () => {
+  const lngDetector = new LngDetector()
+  lngDetector.addDetector({
+    name: 'userSetting',
+    lookup(options) {
+      const { lang = null } = JSON.parse(document.getElementById('user-context-hydrate').textContent || '{}')
+      return lang
     },
+    cacheUserLanguage(lng, options) {},
   })
 
-export default i18n
+  i18n
+    .use(lngDetector)
+    .use(reactI18nextModule)
+    .init({
+      fallbackLng: 'en',
+      debug: true,
```

Comment:
[IMO] I believe this is out of scope of the PR. But it'd be nice to hide debug logs in production.
```js
debug:  process.env.NODE_ENV && (process.env.NODE_ENV === 'development')
```
https://github.com/crowi/crowi/blob/master/tools/webpack.client.js#L81-L85

Reply:
`process.env.NODE_ENV` is defined only in the production environment.
But since it does not seem to be used everywhere, I'll changes it with this pull request.